core/controllers/admin.py

```diff
@@ -111,88 +224,67 @@ def get(self):
     @acl_decorators.can_access_admin_page
     def post(self):
         """Handles POST requests."""
+        action = self.normalized_payload.get('action')
         try:
             result = {}
-            if self.payload.get('action') == 'reload_exploration':
-                exploration_id = self.payload.get('exploration_id')
+            if action == 'reload_exploration':
+                exploration_id = self.normalized_payload.get('exploration_id')
                 self._reload_exploration(exploration_id)
-            elif self.payload.get('action') == 'reload_collection':
-                collection_id = self.payload.get('collection_id')
+            elif action == 'reload_collection':
+                collection_id = self.normalized_payload.get('collection_id')
                 self._reload_collection(collection_id)
-            elif self.payload.get('action') == 'generate_dummy_explorations':
-                num_dummy_exps_to_generate = self.payload.get(
+            elif action == 'generate_dummy_explorations':
+                num_dummy_exps_to_generate = self.normalized_payload.get(
                     'num_dummy_exps_to_generate')
-                num_dummy_exps_to_publish = self.payload.get(
+                num_dummy_exps_to_publish = self.normalized_payload.get(
                     'num_dummy_exps_to_publish')
-                if not isinstance(num_dummy_exps_to_generate, int):
-                    raise self.InvalidInputException(
-                        '%s is not a number' % num_dummy_exps_to_generate)
-                elif not isinstance(num_dummy_exps_to_publish, int):
-                    raise self.InvalidInputException(
-                        '%s is not a number' % num_dummy_exps_to_publish)
-                elif num_dummy_exps_to_generate < num_dummy_exps_to_publish:
+
+                if num_dummy_exps_to_generate < num_dummy_exps_to_publish:
                     raise self.InvalidInputException(
                         'Generate count cannot be less than publish count')
                 else:
                     self._generate_dummy_explorations(
                         num_dummy_exps_to_generate, num_dummy_exps_to_publish)
-            elif self.payload.get('action') == 'clear_search_index':
+            elif action == 'clear_search_index':
```

Comment:
Here clear_search_index is expected to be the value of action key of payload, but payload is not received while clearing the search index from frontend.